### ISSUE TITLE
ConfirmDialog: JAWS Screen reader does not read the text itself

### ISSUE THREAD

**Issue by mshumako** [primefaces/primefaces#8109]
**Describe the defect**
We are using JAWS screen reader https://www.freedomscientific.com/products/software/jaws/
Professional Edition. Version: 2021.2107.12 ILM

When a confirm dialog appears, the reader only reads the "confirm dialog" words and description of the "No" button (it is under focus)
The reader does not read the text itself within the dialogue ("Are you sure you want to proceed?").
A blind or visually impaired user does not receive information about what he/she needs to confirm or reject.

**Reproducer**
You can easily verify this using the showcase:
https://www.primefaces.org/showcase/ui/overlay/confirmDialog.xhtml?jfwid=b39d7

**Environment:**
 - PF Version: v11.0.0-RC2
 - Affected browsers: Chrome

**To Reproduce**
1. Use tab to navigate to the blue "Confirm" button
2. Press enter
3. Screen reader doesn't read the confirm dialog message ("Are you sure you want to proceed?")

**Expected behavior**
Screen reader should read the confirm dialog message.

**Comment by melloware** [comment]
@mshumako Do you have a suggested fix?  

I think this issue is this: https://stackoverflow.com/questions/28112944/jaws-screen-reader-misreads-confirm-dialog-message-ie11

We use aria-labelledBy to point to the title of the dialog.

**Comment by mshumako** [comment]
Seems that ScreenReader reads only the `aria-labelledBy` and ignores the `aria-describedby`. Since the `aria-labelledBy` attribute can accept multiple space separated ids to create an announcement then for my project it will be OK to have something like this: `aria-labelledby="j_idt304:j_idt307_title j_idt304:j_idt307_content"`. Thus, the reader will read both the title and the content.

**Comment by mshumako** [comment]
Can you, please, show me where(,java or .js file) these attributes(`aria-labelledBy` and `aria-describedby`) are added? As far as I understand, this does not happen in [ConfirmDialogRenderer](https://github.com/primefaces/primefaces/blob/master/primefaces/src/main/java/org/primefaces/component/confirmdialog/ConfirmDialogRenderer.java)

**Comment by melloware** [comment]
Its in dialog.js

```javascript
/**
     * Applies all `ARIA` attributes to the contents of this dialog.
     * @protected
     */
    applyARIA: function() {
        this.jq.attr({
            'role': 'dialog'
            ,'aria-describedby': this.id + '_content'
            ,'aria-hidden': !this.cfg.visible
            ,'aria-modal': this.cfg.modal
        });
        
        // GitHub #4727
        var title = this.id + '_title';
        if ($(PrimeFaces.escapeClientId(title)).length) {
            this.jq.attr('aria-labelledby', title);
        }

        this.titlebar.children('a.ui-dialog-titlebar-icon').attr('role', 'button');
    },
```

**Comment by melloware** [comment]
Here is the proper example: https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/alertdialog.html

Let me submit a PR.

**Comment by melloware** [comment]
It looks like we are following ARIA 1.1 so is it possible its a JAWS issue?  Have you tried other screen readers?

**Comment by melloware** [comment]
OK so I corrected the role to be `alertdialog` but other than that we are following all the correct ARIA guidelines.

**Comment by mshumako** [comment]
Thank you
